cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/model/RandomCluster.java

```diff
@@ -377,14 +377,19 @@ private static void markBrokenBrokers(ClusterModel cluster,
 
       // Mark the remaining brokers as a broker with bad disk(s).
       int remainingBrokerWithBadDiskIndex = 0;
-      while (numBrokersWithBadDisk - markedBrokersContainingExcludedTopicReplicas - remainingBrokerWithBadDiskIndex > 0) {
-        Broker brokerToMark = cluster.broker(remainingBrokerWithBadDiskIndex);
+      for (Broker brokerToMark : cluster.brokers()) {
         if (!brokerToMark.replicas().isEmpty() && brokerToMark.isAlive() && !brokerToMark.hasBadDisks()) {
           // Mark one (random) replica on this broker as offline.
           brokerToMark.replicas().iterator().next().markOriginalOffline();
           cluster.setBrokerState(remainingBrokerWithBadDiskIndex, Broker.State.BAD_DISKS);
           remainingBrokerWithBadDiskIndex++;
         }
+        if (numBrokersWithBadDisk - markedBrokersContainingExcludedTopicReplicas == remainingBrokerWithBadDiskIndex) {
```

Comment:
Looks like this if statement should be put at the beginning of each for loop. consider the scenario that `markedBrokersContainingExcludedTopicReplicas`  already equal `numBrokersWithBadDisk` before entering this for loop, in the first iteration of for loop, `remainingBrokerWithBadDiskIndex` may become 1 and actually have more broken-disk broker and `break` will never trigger.